test/common/buffer/buffer_speed_test.cc

```diff
@@ -24,6 +33,76 @@ static void bufferCreateEmpty(benchmark::State& state) {
 }
 BENCHMARK(bufferCreateEmpty);
 
+// Test add performance of OwnedImpl vs WatermarkBuffer
+static void bufferVsWatermarkBuffer(benchmark::State& state) {
+  const uint64_t length = state.range(0);
+  const uint64_t high_watermark = state.range(1);
+  const bool use_watermark_buffer = (state.range(2) != 0);
+
+  for (auto _ : state) {
+    UNREFERENCED_PARAMETER(_);
+    std::unique_ptr<Buffer::Instance> buffer;
+    if (use_watermark_buffer) {
+      buffer = std::make_unique<Buffer::WatermarkBuffer>([]() {}, []() {}, []() {});
+      buffer->setWatermarks(high_watermark);
+    } else {
+      buffer = std::make_unique<Buffer::OwnedImpl>();
+    }
+
+    for (uint64_t idx = 0; idx < length; ++idx) {
+      buffer->add("a", 1);
+    }
+  }
+  benchmark::DoNotOptimize(length);
```

Comment:
nit: I don't think this is needed for this benchmark (and probably not in a lot of the other ones where it has been cargo culted)

Reply:
Thanks.  I removed this one, the remaining ones seem correct.